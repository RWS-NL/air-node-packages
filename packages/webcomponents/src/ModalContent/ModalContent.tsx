import React, { FC, memo } from 'react';
import css from './ModalContent.scss';

/**
 * Creates modal content using pre-defined Rijkswaterstaat styling
 * @param props Props to pass to the modal content
 * @example
 * <ModalContent>
 *   <p>{text}</p>
 * </ModalContent>
 */
export const ModalContent: FC<unknown> = props => (
  <div {...props} className={css.content}>
    {props.children}
  </div>
);

export default memo(ModalContent);