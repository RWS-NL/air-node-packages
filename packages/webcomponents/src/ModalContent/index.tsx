import React, { memo } from 'react';
import css from './ModalContent.scss';

/**
 * Creates modal content using pre-defined Rijkswaterstaat styling
 * @param props Props to pass to the modal content
 * @example
 * ```jsx
 * <ModalContent>
 *   <p>{text}</p>
 * </ModalContent>
 * ```
 */
export const ModalContent = memo(props => (
  <div {...props} className={css.content}>
    {props.children}
  </div>
));

export default ModalContent;
