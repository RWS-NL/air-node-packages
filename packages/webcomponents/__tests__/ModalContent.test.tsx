import { ModalContent } from '@src/ModalContent';
import { shallow } from 'enzyme';
import React from 'react';

describe('Component Testing', () => {
  test('should insert children when getting any', () => {
    const text = 'The Answer is 42';

    const modalContent = shallow(
      <ModalContent>
        <p>{text}</p>
      </ModalContent>
    );

    expect(modalContent.prop('children')).toBeDefined();
    expect(modalContent.find('p').text()).toBe(text);
  });
});

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const modalContent = shallow(<ModalContent />);
    expect(modalContent).toMatchSnapshot();
  });

  test('Optional Props', () => {
    const text = 'The Answer is 42';
    const modalContent = shallow(
      <ModalContent>
        <p>{text}</p>
      </ModalContent>
    );
    expect(modalContent).toMatchSnapshot();
  });
});
