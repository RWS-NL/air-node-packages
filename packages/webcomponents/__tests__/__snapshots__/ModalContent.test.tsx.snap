// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Optional Props 1`] = `
<div
  className="content"
>
  <p>
    The Answer is 42
  </p>
</div>
`;

exports[`Snapshot Testing Required Props 1`] = `
<div
  className="content"
/>
`;
